src/register.jl

```diff
@@ -0,0 +1,138 @@
+greaterthan05(x) = x .> 0.5 # used for the image resize step and for binarizing images
+function imrotate_bin(x, r)
+    return greaterthan05(collect(imrotate(x, r, axes(x); method=BSpline(Constant()))))
+end
+function imrotate_bin_nocrop(x, r)
+    return greaterthan05(collect(imrotate(x, r; method=BSpline(Constant()))))
+end
+imrotate_bin_clockwise_radians(x, r) = imrotate_bin(x, r)
+imrotate_bin_counterclockwise_radians(x, r) = imrotate_bin(x, -r)
+imrotate_bin_clockwise_degrees(x, r) = imrotate_bin_clockwise_radians(x, deg2rad(r))
+imrotate_bin_counterclockwise_degrees(x, r) = imrotate_bin_counterclockwise_radians(x, deg2rad(r))
+
+
+
+# Functions used for the SD minimization
+"""
+Pad images by zeros based on the size of the larger of the two images.
+"""
+function pad_images(im1, im2)
+    n = max(size(im1)..., size(im2)...)
+    im_padded = [collect(padarray(im, Fill(0, (n, n), (n, n)))) for im in [im1, im2]]
+    return im_padded[1], im_padded[2]
+end
+
+"""
+Calculate the centroid of a binary image. If 'rounded', return the
+nearest integer.
+"""
+function compute_centroid(im; rounded=false)
+    xi = 0
+    yi = 0
+    R = sum(im .> 0)
+    n, m = size(im)
+    for ii in range(1, n)
+        for jj in range(1, m)
+            if im[ii, jj] > 0
+                xi += ii
+                yi += jj
+            end
+        end
+    end
+
+    x0, y0 = sum(xi) / R, sum(yi) / R
+    if rounded
+        return round(Int32, x0), round(Int32, y0)
+    else
+        return x0, y0
+    end
+end
+
+"""
+Align images by selecting and cropping so that r1, c1 and r2, c2 are the center.
+These values are expected to be the (integer) centroid of the image. These images
+should already be padded so that there is no danger of cutting into the floe shape.
+"""
+function crop_to_shared_centroid(im1, im2)
+    r1, c1 = compute_centroid(im1; rounded=true)
+    r2, c2 = compute_centroid(im2; rounded=true)
+
+    n1, m1 = size(im1)
+    n2, m2 = size(im2)
+    new_halfn = minimum([minimum([r1, n1 - r1]), minimum([r2, n2 - r2])])
+    new_halfm = minimum([minimum([c1, m1 - c1]), minimum([c2, m2 - c2])])
```

Comment:
Consider using broadcasting:
```suggestion
    new_halfn = minimum(minimum.([r1, n1 - r1], [r2, n2 - r2]))
    new_halfm = minimum(minimum.([c1, m1 - c1], [c2, m2 - c2]))
```

Reply:
I think we need some direct test cases of this function before we start modifying the functions – this is an okay change, but I don't think we have the testing yet for refactoring.

Reply:
It's a different way to do the same work. Would this suffice?
```julia
julia> old(r1, n1, r2, n2) = minimum([minimum([r1, n1 - r1]), minimum([r2, n2 - r2])])
old (generic function with 1 method)

julia> new(r1, n1, r2, n2) = minimum(minimum.([[r1, n1 - r1], [r2, n2 - r2]]))
new (generic function with 1 method)

julia> r1, n1, r2, n2 = rand(1:100_000, 4);

julia> @assert old(r1, n1, r2, n2) == new(r1, n1, r2, n2)

julia>

Reply:
I mean more tests of the `crop_to_shared_centroid` – just to be sure we're not inadvertantly introducing a regression when we change things.

Reply:
```julia
julia> @assert old(r1, n1, r2, n2) == new(r1, n1, r2, n2) == minimum([r1, n1 - r1, r2, n2 - r2])

julia>